lib/puppet/util/windows/registry.rb

```diff
@@ -65,6 +65,30 @@ def values(key)
       vals
     end
 
+    # Retrieve a set of values from a registry key given their names
+    # Value names listed but not found in the registry will not be added to the
+    # resultant Hashtable
+    #
+    # @param key [RegistryKey] An open handle to a Registry Key
+    # @param names [String[]] An array of names of registry values to return if they exist
+    # @return [Hashtable<String, Object>] A hashtable of all of the found values in the registry key
+    def values_by_name(key, names)
+      vals = {}
+      names.each do |name|
+        type, data = nil, nil
+        FFI::Pointer.from_string_to_wide_string(name) do |subkeyname_ptr|
+          begin
+            type, data = read(key, subkeyname_ptr)
```

Comment:
Can probably change this to `_data`

Reply:
No need even for intermediate variable... squashed it down.

Thanks!